hera_mc/weather.py

```diff
@@ -0,0 +1,233 @@
+
+import numpy as np
+from astropy.time import Time
+from math import floor
+from sqlalchemy import Column, BigInteger, Float, String
+
+import tornado.gen
+
+from . import MCDeclarativeBase
+
+katportal_url = 'http://portal.mkat.karoo.kat.ac.za/api/client'
+
+# These are the weather measurements that can be added to the M&C database.
+# To add a sensor, add a similar entry in this dict (paying particular attention
+# to data volume and reduction strategies and periods)
+# Also please add the sensor information to docs/weather_sensor_details.txt
+# You can get the needed information using the katportal example scripts at:
+#   https://github.com/ska-sa/katportalclient/tree/master/examples
+weather_sensor_dict = {'wind_speed': {'sensor_name': 'anc_mean_wind_speed',
+                                      'units': 'm/s',
+                                      'reduction': 'mean', 'period': 60,
+                                      'description': "Mean of  ['wind.wind-speed', 'weather.wind-speed'] in (600 * 1.0s) window (report period: 1s)"},
+                       'wind_gust': {'sensor_name': 'anc_gust_wind_speed',
+                                     'units': 'm/s',
+                                     'reduction': 'max', 'period': 60,
+                                     'description': "Max of  ['wind.wind-speed', 'weather.wind-speed'] in (3 * 1.0s) window (report period: 1s)"},
+                       'wind_direction': {'sensor_name': 'anc_wind_wind_direction',
+                                          'units': 'deg', 'reduction': None,
+                                          'description': "Wind direction angle (report period: 10s)"},
+                       'temperature': {'sensor_name': 'anc_weather_temperature',
+                                       'units': 'deg Celsius', 'reduction': None,
+                                       'description': "Air temperature (report period: 10s)"},
+                       'humidity': {'sensor_name': 'anc_weather_humidity',
+                                    'units': 'percent',
+                                    'reduction': 'mean', 'period': 60,
+                                    'description': "Air humidity (report period: 10s)"},
+                       'pressure': {'sensor_name': 'anc_weather_pressure', 'units': 'mbar',
+                                    'reduction': 'mean', 'period': 60,
+                                    'description': "Barometric pressure (report period: 10s)"},
+                       'rain': {'sensor_name': 'anc_weather_rain', 'units': 'mm',
+                                'reduction': 'sum', 'period': 60,
+                                'description': "Rainfall (report period: 10s)"}}
+
+
+def _reduce_time_vals(times, vals, period, strategy='decimate'):
+    if not isinstance(period, (int, np.int)):
+        raise ValueError('period must be an integer')
+
+    times_keep, inds = np.unique(np.floor(times / period) * period, return_index=True)
```

Comment:
Not a big deal but `times // period` is a slightly more idiomatic way to get the floor division. (The `//` operator in Python is even called `__floordiv__`.)

Reply:
ok, I changed it and added a comment so I'll know what it means later :)